advanced_source/static_quantization_tutorial.rst

```diff
@@ -0,0 +1,675 @@
+
+
+.. code:: python
+
+    def prepare_data_loaders(data_path):  
+
+        traindir = os.path.join(data_path, 'train') 
+        valdir = os.path.join(data_path, 'val') 
+        normalize = transforms.Normalize(mean=[0.485, 0.456, 0.406],  
+                                         std=[0.229, 0.224, 0.225]) 
+
+        dataset = torchvision.datasets.ImageFolder( 
+            traindir, 
+            transforms.Compose([  
+                transforms.RandomResizedCrop(224),  
+                transforms.RandomHorizontalFlip(),  
+                transforms.ToTensor(),  
+                normalize,  
+            ])) 
+
+        dataset_test = torchvision.datasets.ImageFolder(  
+            valdir, 
+            transforms.Compose([  
+                transforms.Resize(256), 
+                transforms.CenterCrop(224), 
+                transforms.ToTensor(),  
+                normalize,  
+            ])) 
+
+        train_sampler = torch.utils.data.RandomSampler(dataset) 
+        test_sampler = torch.utils.data.SequentialSampler(dataset_test) 
+
+        data_loader = torch.utils.data.DataLoader(  
+            dataset, batch_size=train_batch_size, 
+            sampler=train_sampler)  
+
+        data_loader_test = torch.utils.data.DataLoader( 
+            dataset_test, batch_size=eval_batch_size, 
+            sampler=test_sampler) 
+
+        return data_loader, data_loader_test  
+
+
+Next, we'll load in the pre-trained MobileNetV2 model. We provide the URL to download the data from in ``torchvision``  
+`here <https://github.com/pytorch/vision/blob/master/torchvision/models/mobilenet.py#L9>`_. 
+
+.. code:: python
+
+    data_path = 'data/imagenet_1k'  
+    saved_model_dir = 'data/' 
+    float_model_file = 'mobilenet_pretrained_float.pth' 
+    scripted_float_model_file = 'mobilenet_quantization_scripted.pth' 
+    scripted_quantized_model_file = 'mobilenet_quantization_scripted_quantized.pth' 
+
+    train_batch_size = 30 
+    eval_batch_size = 30  
+
+    data_loader, data_loader_test = prepare_data_loaders(data_path) 
+    criterion = nn.CrossEntropyLoss() 
+    float_model = load_model(saved_model_dir + float_model_file).to('cpu')  
+ 
+    # Next, we'll "fuse modules"; this can both make the model faster by saving on memory access  
+    # while also improving numerical accuracy. While this can be used with any model, this is 
+    # especially common with quantized models.  
+
+    print('\n Inverted Residual Block: Before fusion \n\n', float_model.features[1].conv) 
+    float_model.eval()  
+
+    # Fuses modules 
+    float_model.fuse_model()  
+
+    # Note fusion of Conv+BN+Relu and Conv+Relu 
+    print('\n Inverted Residual Block: After fusion\n\n',float_model.features[1].conv)  
+
+  
+Finally to get a "baseline" accuracy, let's see the accuracy of our un-quantized model  
+with fused modules  
+
+.. code:: python
+
+    num_eval_batches = 10 
```

Comment:
Let us go with num_eval_batches = 1000 and eval_batch_size of 50, to correspond to full imagenet validation dataset size.